app/models/service_template_transformation_plan/validate_config_info.rb

```diff
@@ -9,31 +9,94 @@ module ClassMethods
+      # ansible playbooks validation
+      # ------------------------------------------------------------------------
       pre_service_id  = config_info[:pre_service].try(:id) || config_info[:pre_service_id]
       post_service_id = config_info[:post_service].try(:id) || config_info[:post_service_id]
 
+      # Add as a part of validating migration plan
+      # check service_type. Filter type=ServiceTemplateAnsiblePlaybook
+      if pre_service_id.present?
+        pre_service = ServiceTemplate.find(pre_service_id)
+        unless pre_service.nil?
+          raise _('Premigration service type MUST be \"ServiceTemplateAnsiblePlaybook\"') if pre_service.type != "ServiceTemplateAnsiblePlaybook"
+        end
+      end
+
+      if post_service_id.present?
+        post_service = ServiceTemplate.find(post_service_id)
+        unless post_service.nil?
+          raise _('Postmigration service type MUST be \"ServiceTemplateAnsiblePlaybook\"') if post_service.type != "ServiceTemplateAnsiblePlaybook"
+        end
+      end
+
+      # ------------------------------------------------------------------------
+      # VMs validation
+      # ------------------------------------------------------------------------
       vms = []
       if config_info[:actions]
         vm_objects = VmOrTemplate.where(:id => config_info[:actions].collect { |vm_hash| vm_hash[:vm_id] }.compact).index_by(&:id).stringify_keys
         config_info[:actions].each do |vm_hash|
           vm_obj = vm_objects[vm_hash[:vm_id]] || vm_hash[:vm]
           next if vm_obj.nil?
+          # vm_obj.invalid? added as a part of validating migration plan
+          raise _("Invalid VM found #{vm_obj.name}") if vm_obj.invalid?
 
           vm_options = {}
           vm_options[:pre_ansible_playbook_service_template_id] = pre_service_id if vm_hash[:pre_service]
           vm_options[:post_ansible_playbook_service_template_id] = post_service_id if vm_hash[:post_service]
-          vm_options[:osp_security_group_id] = vm_hash[:osp_security_group_id] if vm_hash[:osp_security_group_id].present?
-          vm_options[:osp_flavor_id] = vm_hash[:osp_flavor_id] if vm_hash[:osp_flavor_id].present?
           vm_options[:cpu_right_sizing_mode] = vm_hash[:cpu_right_sizing_mode] if vm_hash[:cpu_right_sizing_mode].present?
           vm_options[:memory_right_sizing_mode] = vm_hash[:memory_right_sizing_mode] if vm_hash[:memory_right_sizing_mode].present?
+          vm_options[:osp_flavor_id] = vm_hash[:osp_flavor_id] if vm_hash[:osp_flavor_id].present?
+          vm_options[:osp_security_group_id] = vm_hash[:osp_security_group_id] if vm_hash[:osp_security_group_id].present?
+
+          # verify the vm flavor belongs to the openstack tenant/project. added as a part of validating migration plan
+          if osp_flavors.present? && vm_options[:osp_flavor_id].present?
+            vm_flavor = Flavor.find(vm_options[:osp_flavor_id])
+            if !vm_flavor.nil?
```

Comment:
Minor nit, rather than a double negative, you could do `if vm_flavor.present?`.

Reply:
Will change it to `vm.flavor.present?`.  Though I did not like the double negative, still used it because it was so clear in showing the intent.